tech-specs/src/explore/design/ledger/pos-integration.md

```diff
@@ -0,0 +1,257 @@
+
+- `parameters`: the system parameters
+- `parameters/{parameters_hash}`: the system parameters value with this `parameters_hash `
+- `parameters/{parameters_hash}/{validator_address}`: a set of validators who want to switch to parameters with this `parameters_hash` (only keys are needed here, so the value can be a `unit`)
+- `validator/{validator_address}/consensus_key`
+- `validator/{validator_address}/state`
+- `validator/{validator_address}/total_deltas`
+- `validator/{validator_address}/voting_power`
+- `slash/{validator_address}`: a list of slashes, where each record contains epoch and slash rate
+- `bond/{bond_source}/{bond_validator}/delta`
+- `unbond/{unbond_source}/{unbond_validator}/deltas`
+- `validator_set/active`
+- `validator_set/inactive`
+- `total_voting_power`
+
+- standard validator metadata (these are regular storage values, not epoched data):
+  - `validator/{validator_address}/staking_reward_address`: an address that should receive staking rewards, if not set, the rewards are sent to `validator_address`
+  - TBA (e.g. alias, website, description, delegation commission rate, etc.)
+
+Additionally, only XAN tokens can be staked. The tokens being staked (bonds and unbonds amounts) are kept in the XAN token account under `{xan_address}/balance/{pos_address}`.
+
+## Initialization
+
+The PoS system is initialized via a native VP interface that is given the validator set for the genesis block.
+
+## Staking rewards and transaction fees
+
+Staking rewards for validators are rewarded in Tendermint's method `BeginBlock` in the base ledger. If a validator has specified `validator/{validator_address}/staking_reward_address` the rewards are credited to this address, otherwise, the validator's address is used.
+
+To a validator who proposed a block (`block.header.proposer_address`), the system rewards tokens based on the `block_proposer_reward` PoS parameter and each validator that voted on a block (`block.last_commit_info.validator` who `signed_last_block`) receives `block_vote_reward`.
+
+All the fees that are charged in a transaction execution (DKG transaction wrapper fee and transactions applied in a block) are transferred into a fee pool, which is another special account controlled by the PoS module.
+
+- TODO describe the fee pool, related to <https://github.com/anomanetwork/anoma/issues/48>, <https://github.com/anomanetwork/anoma/issues/51> and <https://github.com/anomanetwork/anoma/issues/72>
+
+## Transactions
+
+The transactions are assumed to be applied in epoch `n`. Any transaction that modifies [epoched data](/explore/design/pos.md#epoched-data) updates the structure as described in [epoched data storage](/explore/design/pos.md#storage).
+
+For transactions that burn tokens, we implement a ["token burner" account](/explore/design/ledger/vp.md#token-burner-vp). Burnt tokens should be credited to this account and no token can be ever be debited by anyone.
```

Comment:
Just as a note, it's possible that we may want to redistribute burned tokens or do something other than burn them, to be determined. Ideally we could keep the system flexible.

Reply:
We implemented this as a "PoS slash pool", which receives slashed tokens, but nobody can withdraw from it